dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/hooks/SmsCommandObjectBundleHook.java

```diff
@@ -108,13 +114,23 @@ private void process( SMSCommand command )
 
     private void getReferences( SMSCommand command )
     {
+        CategoryOptionCombo defaultCoc = categoryService.getDefaultCategoryOptionCombo();
+
         command.getCodes().stream()
-            .filter( c -> c.hasDataElement() )
-            .forEach( c -> c.setDataElement( dataElementService.getDataElement( c.getDataElement().getUid() ) ) );
+            .filter( SMSCode::hasDataElement )
+            .forEach( c -> {
+                c.setOptionId( c.getOptionId() == null ? defaultCoc
+                    : categoryService.getCategoryOptionCombo( c.getOptionId().getUid() ) );
```

Comment:
I don't know how many times we will perform this query, might it rather be worth collecting all optionIds and loading them all with one query using `in`? 
If it doesn't affect performance it might not be worth it. You know it :)